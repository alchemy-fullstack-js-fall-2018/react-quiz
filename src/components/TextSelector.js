import React from 'react';

export default function TextSelector({
  text,
  onChange
}) {
  return (
    <div>
      <label name="text">Text:</label>
      <input
        name="text"
        type="text"
        value={text}
        onChange={onChange}
      />
    </div>
  );
}
