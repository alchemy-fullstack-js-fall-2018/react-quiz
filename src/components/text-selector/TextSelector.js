import React, { Fragment } from 'react';

const TextSelector = ({ text, handleChange }) => {
  return (
    <Fragment>
      <input  name="text" value={text} onChange={handleChange}/>
    </Fragment>
  );
};

export default TextSelector;

