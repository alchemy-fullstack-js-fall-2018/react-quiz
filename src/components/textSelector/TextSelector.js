import React, { Fragment } from 'react';

const TextSelector = ({ updateText }) => {
  return (
    <Fragment>
      <label htmlFor='textBox'>Input Text Here</label>
      <input name='textBox' id='textBox' type='text' onChange={updateText} />
    </Fragment>
  );
};

export default TextSelector;
