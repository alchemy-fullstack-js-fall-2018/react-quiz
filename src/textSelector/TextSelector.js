import React from 'react';

const TextSelector = (body) => {
  const container = body;
  return (
    <label>Enter Text here</label>
    <input>{container}</input>
  );
}

export default TextSelector;
