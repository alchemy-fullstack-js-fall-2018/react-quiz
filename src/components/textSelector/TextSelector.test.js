import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('text selector', () => {
  let wrapper;
  beforeEach(() => {
    const onChangeFn = jest.fn();
    wrapper = shallow(
      <TextSelector
        onChange={onChangeFn}
        text={'asdf'}
      />
    );
  });

  it('renders a text selector', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
