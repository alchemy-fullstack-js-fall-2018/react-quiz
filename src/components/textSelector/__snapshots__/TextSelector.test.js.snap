// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders an TextSelector component 1`] = `
<Fragment>
  <label
    htmlFor="textBox"
  >
    Input Text Here
  </label>
  <input
    id="textBox"
    name="textBox"
    type="text"
  />
</Fragment>
`;
